Java/core/src/main/java/com/amazon/randomcutforest/util/WeightedIndex.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.randomcutforest.util;
+
+import static com.amazon.randomcutforest.CommonUtils.checkArgument;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+/**
+ * a container class that manages weights
+ * 
+ * @param <Q>
+ */
+public class WeightedIndex<Q> {
```

Comment:
Since there is no type restriction on Q, I would just call this `Weighted<Q>`.

Reply:
Done.